File: src/Layer.java
import java.awt.image.BufferedImage;
import java.io.File;

import javax.imageio.ImageIO;

public class Layer {
	final static int NO_TYPE=0,
			TYPE_HEAD=1,
			TYPE_NECK=2,
			TYPE_BODY=3,
			TYPE_HAND_TOP=4,
			TYPE_HAND_BOTTOM=5,
			TYPE_LEG_TOP=6,
			TYPE_LEG_BOTTOM=7,
			TYPE_WEAPON=8;
	int x,y,scaledWidth,scaledHeight,basicWidth,basicHeight,xsize,xwidth,xheight;
	double rotationAngle;
	BufferedImage basicImage;
	String imageName,layerName;
	public Layer(String imageName, int x, int y, int width, int height, int xsize, double angle){
		this.x=x;
		this.y=y;
		this.rotationAngle=angle;
		this.imageName=imageName;
		this.layerName=this.imageName.substring(0,this.imageName.length());
		try{
			this.basicImage=ImageIO.read(new File("./drawable/"+imageName));
			this.basicWidth=this.basicImage.getWidth();
			this.basicHeight=this.basicImage.getHeight();
			this.scaledWidth=width;
			this.scaledHeight=height;
			this.xsize=xsize;
			this.xwidth=this.scaledWidth*100/this.xsize*100/this.basicWidth;
			this.xheight=this.scaledWidth*100/this.xsize*100/this.basicWidth;
		}
		catch(Exception ex){
			ex.printStackTrace();
		}
	}
	public Layer(String imageName){
		this.x=0;
		this.y=0;
		this.rotationAngle=0;
		this.imageName=imageName;
		this.layerName=this.imageName;
		try{
			this.basicImage=ImageIO.read(new File("./drawable/"+imageName));
			this.basicWidth=this.basicImage.getWidth();
			this.basicHeight=this.basicImage.getHeight();
			this.scaledWidth=this.basicImage.getWidth();
			this.scaledHeight=this.basicImage.getHeight();
			this.xsize=100;
			this.xwidth=100;
			this.xheight=100;
		}
		catch(Exception ex){
			ex.printStackTrace();
		}
	}
	public void updateSize(){
		scaledWidth = basicWidth*xsize*xwidth/10000;
		scaledHeight = basicHeight*xsize*xheight/10000;
	}
}

File: src/logic/Layer.kt
package logic

import java.awt.image.BufferedImage
import java.io.File
import java.io.Serializable

import javax.imageio.ImageIO

class Layer (var imageName: String, var x : Float = 0f, var y : Float = 0f, var scale : Float = 1f, var scaleX : Float = 1f,
             var scaleY : Float = 1f, var angle : Float = 0f) : Serializable{
    var basicWidth: Int = 0
    var basicHeight: Int = 0
    var layerName: String
    @Transient
    var basicImage: BufferedImage? = null

    init {
        layerName = imageName.substring(0, imageName.length - 4)
        loadImage()
    }
    constructor(origin : Layer) : this(origin.imageName, origin.x, origin.y, origin.scale, origin.scaleX, origin.scaleY,
            origin.angle)

    fun loadImage() {
        try {
            basicImage = ImageIO.read(File("./drawable/$imageName"))
            basicWidth = basicImage!!.width
            basicHeight = basicImage!!.height
        } catch (ex: Exception) {
            ex.printStackTrace()
        }
    }
}
